starlette/middleware/base.py

```diff
@@ -42,18 +48,42 @@ async def coro() -> None:
         task = loop.create_task(coro())
         message = await queue.get()
         if message is None:
+            queue.task_done()
             task.result()
             raise RuntimeError("No response returned.")
         assert message["type"] == "http.response.start"
 
         async def body_stream() -> typing.AsyncGenerator[bytes, None]:
-            while True:
+            def streaming_predicate(
+                msg: typing.Optional[dict], more_body: bool = True
```

Comment:
`more_body` seems to always be `True`. Why is it made an argument?

Reply:
Ah, a holdover from when I was originally experimenting with separating out some code on it. I'll pull that out.